test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetRestoreCommandTest.cs

```diff
@@ -2220,10 +2220,88 @@ var targetFileContents
                     waitForExit: true);
 
                 // Assert
-                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);               
+                Assert.True(0 == r.Item1, r.Item2 + " " + r.Item3);
                 var packageFileA = Path.Combine(randomSolutionFolder, "GlobalPackages", "packagea","1.1.0", "packageA.1.1.0.nupkg");
                 Assert.True(File.Exists(packageFileA));
             }
         }
+
+        [Fact]
+        public void RestoreCommand_WithAuthorSignedPackage_Succeeds()
+        {
+            using (var packageSourceFolder = TestDirectory.Create())
+            using (var packageDestinationFolder = TestDirectory.Create())
+            using (var projectFolder = TestDirectory.Create())
+            {
+                var packageFile = new FileInfo(Path.Combine(packageSourceFolder.Path, "TestPackage.AuthorSigned.1.0.0.nupkg"));
+                var package = GetResource(packageFile.Name);
+
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                var projectFile = new FileInfo(Path.Combine(projectFolder, "ClassLibrary1.csproj"));
+                File.WriteAllText(
+                    projectFile.FullName,
+                    @"<?xml version=""1.0"" encoding=""utf-8""?>
+<Project ToolsVersion=""4.0"" DefaultTargets=""Build"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+  <PropertyGroup>
+    <Configuration Condition="" '$(Configuration)' == '' "">Debug</Configuration>
+    <Platform Condition="" '$(Platform)' == '' "">AnyCPU</Platform>
+    <ProjectGuid>{8586D895-886A-41C9-AAE0-B5510BFA50FC}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <RootNamespace>ClassLibrary1</RootNamespace>
+    <AssemblyName>ClassLibrary1</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+  </PropertyGroup>
+  <PropertyGroup Condition="" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' "">
+    <OutputPath>bin\Debug\</OutputPath>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include=""System"" />
+    <Reference Include=""System.Core"" />
+    <Reference Include=""Microsoft.CSharp"" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include=""TestPackage.AuthorSigned"">
```

Comment:
what is the source of this package?

Reply:
It's a static resource [included in this PR](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-d708b4762ffd5d9e1e514cffb32bf6d2).

This test first [extracts the static resource to disk](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-efde37b61c6a1135b7b8f64f080c4d28R2239), then [passes the directory as a package source](https://github.com/NuGet/NuGet.Client/pull/2074/files#diff-efde37b61c6a1135b7b8f64f080c4d28R2280).